plugins/affiliation/src/main.rs

```diff
@@ -0,0 +1,573 @@
+				Error::UnspecifiedQueryState
+			})?;
+		let commit_view: CommitContributorView = serde_json::from_value(view_value)
+			.map_err(|_| Error::UnexpectedPluginQueryInputFormat)?;
+
+		match affiliation.affiliated_type {
+			AffiliatedType::Author => {
+				contributors.insert((commit_view.author.name, commit_view.author.email));
+			}
+			AffiliatedType::Committer => {
+				contributors.insert((commit_view.committer.name, commit_view.committer.email));
+			}
+			AffiliatedType::Neither => (),
+			// Add both author and committer to the hash set if both are affiliated, in case they are distinct
+			AffiliatedType::Both => {
+				contributors.insert((commit_view.author.name, commit_view.author.email));
+				contributors.insert((commit_view.committer.name, commit_view.committer.email));
+			}
+		};
+	}
+
+	// Add string representation of affiliated contributor with count of associated commits
+	for contributor in contributors.into_iter() {
+		let count = count_commits_for(engine, repo.clone(), contributor.1).await?;
+		contributor_freq_map.insert(contributor.0, count);
+	}
+
+	// Get the total number of affiliated contributors
+	let count = contributor_freq_map.keys().count() as i64;
+
+	// Add each contributor-count pair as a concern
+	for (contributor, count) in contributor_freq_map.into_iter() {
+		let concern = format!("Contributor {} has count {}", contributor, count);
+		engine.record_concern(concern);
+	}
+
+	log::info!("completed affiliation metric");
+
+	Ok(count)
+}
+
+/// Gets the number of commits to a repo associated with a given contributor
+async fn count_commits_for(
+	engine: &mut PluginEngine,
+	repo: LocalGitRepo,
+	email: String,
+) -> Result<i64> {
+	let contributor_repo = DetailedGitRepo {
+		local: repo,
+		details: Some(email.clone()),
+	};
+	let contributor_value = engine
+		.query("mitre/git/commits_for_contributor", contributor_repo)
+		.await
+		.map_err(|e| {
+			log::error!("failed to get commits for contributor {}: {}", email, e);
+			Error::UnspecifiedQueryState
+		})?;
+
+	let contributor_view: ContributorView = serde_json::from_value(contributor_value)
+		.map_err(|_| Error::UnexpectedPluginQueryInputFormat)?;
+	Ok(contributor_view.commits.len() as i64)
+}
+
+#[derive(Clone, Debug)]
+struct AffiliationPlugin;
+
+impl Plugin for AffiliationPlugin {
+	const PUBLISHER: &'static str = "mitre";
+
+	const NAME: &'static str = "affiliation";
+
+	fn set_config(&self, config: Value) -> StdResult<(), ConfigError> {
+		let conf =
+			serde_json::from_value::<Config>(config).map_err(|e| ConfigError::Unspecified {
+				message: e.to_string(),
+			})?;
+		CONFIG.set(conf).map_err(|_e| ConfigError::Unspecified {
+			message: "config was already set".to_owned(),
+		})
+	}
+
+	fn default_policy_expr(&self) -> Result<String> {
+		match CONFIG
+			.get()
+			.ok_or_else(|| {
+				log::error!("tried to access config before set by Hipcheck core!");
+				Error::UnspecifiedQueryState
+			})?
+			.count_threshold
```

Comment:
On my side clippy marks this:

```
warning: in a `match` scrutinee, avoid complex blocks or closures with blocks; instead, move the block or closure higher and bind it with a `let`
   --> plugins/affiliation/src/main.rs:368:19
    |
368 |               .ok_or_else(|| {
    |  ____________________________^
369 | |                 log::error!("tried to access config before set by Hipcheck core!");
370 | |                 Error::UnspecifiedQueryState
371 | |             })?
    | |_____________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#blocks_in_conditions
    = note: `#[warn(clippy::blocks_in_conditions)]` on by default
```

Reply:
Still not marking it on mine. Explains why you had to fix it for the other plugin when you did your latest MR.

Reply:
@mchernicoff maybe do a `rustup update`, there was recently a new version of Rust put out (1.82)

Reply:
Ah hah. I was working on Windows, not WSL like I usually do, to make sure we were OS filesystem agnostic, and I was still on 1.81 there.